pkg/api/validation/events.go

```diff
@@ -25,11 +25,19 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
Ah, right.  Added a comment.

Are we enforing version match between api resource and its reference? i.e. is this valid
{
   kind: "event"
   apiversion: "v1beta1"
   ...
   involvedObj: {
      kind: Node
      apiversion: v1beta3
   }
}

If this is invalid, then things might be eaiser since we can do a inline conversion in the event; otherwise we need to do nested conversion?